Library/Homebrew/cmd/tap.rb

```diff
@@ -17,8 +17,10 @@ def tap
         tap.install :clone_target => ARGV.named[1],
                     :full_clone   => ARGV.include?("--full"),
                     :quiet        => ARGV.quieter?
-      rescue TapAlreadyTappedError => e
-        opoo e
+      rescue TapRemoteMismatchError => e
+        odie e
+      rescue TapAlreadyTappedError, TapAlreadyUnshallowError => e
+        $stderr.puts e unless ARGV.quieter?
```

Comment:
opoo will show it as warning but not change the exit code. I think this is what we want.